trlx/utils/modeling.py

```diff
@@ -241,3 +254,207 @@ def update(self, xs: torch.Tensor) -> Tuple[float, float]:
         self.count = tot_count
 
         return xs_mean, (xs_var * xs_count / (xs_count - 1)).sqrt()
+
+
+# OpenDelta utilities
+
+
+MODIFIED_MODULES_DICT = {
+    "gpt2": {
+        "attention": ["attn.c_attn", "attn.c_proj"],
+        "mlp": ["mlp.c_fc", "mlp.c_proj"],
+        "all": ["attn.c_attn", "attn.c_proj", "mlp.c_fc", "mlp.c_proj"],
+    },
+    "gptj": {
+        "attention": ["attn.q_proj", "attn.k_proj", "attn.v_proj"],
+        "mlp": ["mlp.fc_in", "mlp.fc_out"],
+        "all": [
+            "attn.q_proj",
+            "attn.k_proj",
+            "attn.v_proj",
+            "attn.out_proj",
+            "mlp.fc_in",
+            "mlp.fc_out",
+        ],
+    },
+    "gpt_neox": {
+        "attention": ["attention.query_key_value"],
+        "mlp": ["mlp.dense_h_to_4h", "mlp.dense_4h_to_h"],
+        "all": [
+            "attention.query_key_value",
+            "attention.dense",
+            "mlp.dense_h_to_4h",
+            "mlp.dense_4h_to_h",
+        ],
+    },
+    "opt": {
+        "attention": [
+            "self_attn.k_proj",
+            "self_attn.v_proj",
+            "self_attn.q_proj",
+            "self_attn.out_proj",
+        ],
+        "mlp": ["fc1", "fc2"],
+        "all": [
+            "self_attn.k_proj",
+            "self_attn.v_proj",
+            "self_attn.q_proj",
+            "self_attn.out_proj",
+            "fc1",
+            "fc2",
+        ],
+    },
+    "bloom": {
+        "attention": ["self_attention.query_key_value", "self_attention.dense"],
+        "mlp": ["mlp.dense_h_to_4h", "mlp.dense_4h_to_h"],
+        "all": [
+            "self_attention.query_key_value",
+            "self_attention.dense",
+            "mlp.dense_h_to_4h",
+            "mlp.dense_4h_to_h",
+        ],
+    },
+}
+
+
+def generate_layer_regex(config, num_layers_unfrozen: int = -1) -> str:
```

Comment:
Whats the reason for not wanting to put all of the delta stuff in its own file...